main.js

```diff
@@ -0,0 +1,69 @@
+var towersOfHanoi = {
+  gameboard: [
+    [1],
+    [5, 4, 3, 2],
+    []
+  ],
+  moveDisc: function (onePeg, twoPeg){
+    if(!this.checkIllegal(onePeg, twoPeg)){;
+    this.gameboard[twoPeg-1].push(this.gameboard[onePeg-1].pop());
+    this.printBoard();
+    this.checkWinner();
+  }
+  },
+  checkWinner: function (){
+    if (this.gameboard[0].length === 0 && this.gameboard.filter(function (peg) {
+      return peg.length === towersOfHanoi.gameConditions.numDiscs;
+    }).length === 1) {
+      console.log('You\'re a winner!');
+      this.reset()
+    }
+  },
+  checkIllegal: function (onePeg, twoPeg){
+    if(this.gameboard[onePeg-1][this.gameboard[onePeg-1].length-1] > this.gameboard[twoPeg-1][this.gameboard[twoPeg-1].length-1]){
+      console.log('Illegal move! Think of something else!')
+      return true;
+    }
+  },
+  reset: function (){
+    this.changeFormat(this.gameConditions.numPegs, this.gameConditions.numDiscs);
+    console.log('The game has been reset. Best of luck!')
+    this.printBoard()
+  },
+  changeFormat: function (numPegs, numDiscs) {
+    this.gameboard = [];
+    this.gameConditions.numDiscs = numDiscs;
+    this.gameConditions.numPegs = numPegs;
+    function addArray (){
+      if (towersOfHanoi.gameboard.length === numPegs){
+        return
+      } else {
+        towersOfHanoi.gameboard.push([]);
+        addArray();
+      }
+    };
+    function addDisc (){
+      if (numDiscs === 0){
+        return
+      } else {
+        towersOfHanoi.gameboard[0].push(numDiscs);
+        numDiscs --;
+        addDisc();
+      }
+    }
+    addArray();
+    addDisc();
+    this.printBoard();
+  },
+  printBoard: function () {
+    this.gameboard.map(function (peg) {
```

Comment:
We wanted to print the board like this:

```js
--- 5 4 3 2 1
---
---
```

Also, this is an incorrect way to use `map` as it is not returning anything. A good litmus test here is to ask yourself if it could be replaced with `forEach`.

Reply:
Interesting... I didn't use .forEach() here because I didn't want to alter the source arrays by joining them together. Is the issue that I used a console.log() statement instead of an actual 'return' statement with .map()? Thanks!

Reply:
I think I get it actually, since I'm just logging the entries, .forEach() is fine because console.log() doesn't alter the source array. Pushed another commit with a simpler take on this.printBoard if you want to take a look. Thanks!